lib/sshutils/x11/forward_test.go

```diff
@@ -26,85 +25,62 @@ import (
 func TestForward(t *testing.T) {
 	ctx := context.Background()
 
-	// Create a fake client display. In practice, the display
-	// set in $DISPLAY is used to connect to the client display.
-	fakeClientDisplay, err := net.Listen("tcp", ":0")
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		require.NoError(t, fakeClientDisplay.Close())
-	})
+	// Open a dual sided connection on a new tcp listener
+	openConn := func(t *testing.T) (clientConn net.Conn, serverConn net.Conn) {
+		l, err := net.Listen("tcp", ":0")
+		require.NoError(t, err)
+		t.Cleanup(func() { require.NoError(t, l.Close()) })
 
-	// Handle connections to the client XServer
-	echoMsg := "msg"
-	go func() {
-		for {
-			localConn, err := fakeClientDisplay.Accept()
+		serverErrC := make(chan error)
+		serverConnC := make(chan net.Conn)
+		go func() {
+			serverConn, err := l.Accept()
 			if err != nil {
-				// listener is closed, test is done.
-				return
+				serverErrC <- err
+				close(serverConnC)
 			}
+			serverConnC <- serverConn
+			close(serverErrC)
+		}()
 
-			go func() {
-				defer localConn.Close()
-
-				// read request and expect what was written to server
-				bytes, err := io.ReadAll(localConn)
-				require.NoError(t, err)
-				require.Equal(t, echoMsg, string(bytes))
-			}()
-		}
-	}()
-
-	// Create a fake XServer proxy just like the one in sshserver.
-	sl, serverDisplay, err := OpenNewXServerListener(DefaultDisplayOffset, DefaultMaxDisplays, 0)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		require.NoError(t, sl.Close())
-	})
+		clientConn, err = net.Dial("tcp", l.Addr().String())
+		require.NoError(t, err)
+		t.Cleanup(func() { clientConn.Close() })
 
-	// Handle connection to XServer proxy
-	go func() {
-		for {
-			serverConn, err := sl.Accept()
-			if err != nil {
-				// listener is closed, test is done.
-				return
-			}
+		serverConn = <-serverConnC
+		require.NoError(t, <-serverErrC)
+		t.Cleanup(func() { serverConn.Close() })
 
-			go func() {
-				defer serverConn.Close()
+		return clientConn, serverConn
+	}
 
-				clientConn, err := net.Dial("tcp", fakeClientDisplay.Addr().String())
-				if err != nil {
-					// fakeClientDisplay is closed, test is done.
-					return
-				}
+	cConn1, sConn1 := openConn(t)
+	cConn2, sConn2 := openConn(t)
 
-				clientXConn, ok := clientConn.(*net.TCPConn)
-				require.True(t, ok)
-				defer clientConn.Close()
+	// Start forwarding between connections so that we get
+	// this flow: cConn1 -> sConn1 -> cConn2 -> sConn2.
+	serverConnToForward, ok := sConn1.(*net.TCPConn)
+	require.True(t, ok)
+	clientConnToForward, ok := cConn2.(*net.TCPConn)
+	require.True(t, ok)
 
-				err = Forward(ctx, clientXConn, serverConn)
-				require.NoError(t, err)
-			}()
-		}
+	forwardErrC := make(chan error)
```

Comment:
Should this be buffered with capacity 1, or are we okay with the sender blocking until we read the error at the end of the test?